### PR TITLE
Obtaining an Access Token

### PR DESCRIPTION
**Pull request by ghost**
Changes throughout the section. Moved access token response details to Authorization Server Obligations Section.

[x] NOTE CHANGE NEEDED IN DIAGRAM:  Change expires_in value to 300 or less.

### CODE REVIEW

**Comment by jmandel** [comment]
@bakerdb I added a commit on this PR to fix the diagram as requested :-)

File: authorization.md [hunk 1]
```diff
@@ -69,32 +69,46 @@ key material as integer values). This means that:
...
 #### Protocol details
 
-Before a backend service can request an access token, it must generate a
+Before a backend service can request an access token, it SHALL generate a
 one-time-use JSON Web Token (JWT) that will be used to authenticate the service to
-the EHR's authorization server. The authentication JWT is constructed with the
-following claims, and then signed with the backend service's private RSA key
-(RSA SHA-384 signature). For a practical reference on JWT, as well as debugging
+the EHR's authorization server. The authentication JWT SHALL include the
+following claims, and SHALL be signed with the backend service's private RS-384
```
Comment (jmandel) [inline authorization.md]:
```suggestion
following claims, and SHALL be signed with the backend service's private
```

File: authorization.md [hunk 2]
```diff
@@ -69,32 +69,46 @@ key material as integer values). This means that:
...
 
-Before a backend service can request an access token, it must generate a
+Before a backend service can request an access token, it SHALL generate a
 one-time-use JSON Web Token (JWT) that will be used to authenticate the service to
-the EHR's authorization server. The authentication JWT is constructed with the
-following claims, and then signed with the backend service's private RSA key
-(RSA SHA-384 signature). For a practical reference on JWT, as well as debugging
+the EHR's authorization server. The authentication JWT SHALL include the
+following claims, and SHALL be signed with the backend service's private RS-384 
+or EC384 signature. For a practical reference on JWT, as well as debugging
```
Comment (jmandel) [inline authorization.md]:
```suggestion
key (which SHOULD be an RS384 or EC384 signature). For a practical reference on JWT, as well as debugging
```

**Comment by jmandel** [comment]
@bakerdb I've just cleaned up merge conflicts via https://github.com/smart-on-fhir/fhir-bulk-data-docs/pull/90/commits/3e27ce2770a9fa680b7a68e4315d6ec46ebd16f6. Do you want to review the current state and make sure it looks okay before we merge this into master?

**Comment by ghost** [comment]
I do see a couple of things I would advise to change.  Is there a way for me to edit this, or do I need to send you the changes requested?

-Dixie


Dixie B. Baker, Ph.D.
Senior Partner
Martin, Blanck and Associates
Office (Redondo Beach, CA):  310-791-9671
Mobile:  310-279-2579

On Nov 20, 2018, at 1:06 PM, Josh Mandel <notifications@github.com<mailto:notifications@github.com>> wrote:


@bakerdb<https://github.com/bakerdb> I've just cleaned up merge conflicts via 3e27ce2<https://github.com/smart-on-fhir/fhir-bulk-data-docs/commit/3e27ce2770a9fa680b7a68e4315d6ec46ebd16f6>. Do you want to review the current state and make sure it looks okay before we merge this into master?

**Comment by jmandel** [comment]
You can make any additional edits here in this PR (i.e., by adding commits to your `bakerdb:patch-5` branch). I just added one commit to your branch to preemptively resolve merge conflicts.

File: authorization.md [hunk 3]
```diff
@@ -124,32 +124,46 @@ x-coordinate, and y-coordinate, for EC keys)
...
 
-Before a backend service can request an access token, it must generate a
-one-time-use JSON Web Token (JWT) that will be used to authenticate the service to
-the EHR's authorization server. The authentication JWT is constructed with the
-following claims, and then signed with the backend service's private RSA key
-(RSA SHA-384 signature). For a practical reference on JWT, as well as debugging
+Before a client can request an access token, it SHALL generate a
+one-time-use JSON Web Token (JWT) that will be used to authenticate the client to
+the FHIR authorization server. The authentication JWT SHALL include the
+following claims, and SHALL be signed with the client's private RS-384
```
Comment (jmandel) [inline authorization.md]:
```suggestion
following claims, and SHALL be signed with the client's private
```

Should delete this extraneous "RS-384"

**Pull request merged**